Compiling mujoco with MinGW fails because of redefined stat

### Intro

I'm compiling Mujoco for different targets in a Linux-based CI.

### My setup

- Ubuntu 24.04 amd64
- Mujoco 3.4.0

### What's happening? What did you expect?

Building Mujoco with MinGW fails because `stat` is redefined.
```
[  9%] Building CXX object CMakeFiles/mujoco.dir/src/user/user_resource.cc.obj
/home/user/mujoco/src/user/user_resource.cc:36: error: "stat" redefined [-Werror]
   36 |   #define stat _stat
      | 
In file included from /home/user/mujoco/src/user/user_resource.cc:19:
/usr/share/mingw-w64/include/sys/stat.h:279: note: this is the location of the previous definition
  279 | #define stat _stat64
      | 
cc1plus: all warnings being treated as errors
```
I'd expect that it compiles without errors. This used to work with Mujoco 3.3.3.

As a workaround, I can comment out the define in line 36.

### Steps for reproduction

* Building Mujoco with mingw:
  `cmake -DCMAKE_TOOLCHAIN_FILE=~/cmake-toolchain-mingw64-x86_64.cmake -DCMAKE_COMPILE_WARNING_AS_ERROR=false ..`
* Remove problematic section from `mujoco.rc` file (Issue #839)
* Workaround localtime issue #3037
* Run `make`


This is the content of the toolchain file:
```
set(CMAKE_SYSTEM_NAME Windows)
set(CMAKE_SYSTEM_PROCESSOR x86_64)

find_program(CMAKE_C_LINKER NAMES x86_64-w64-mingw32-ld)
find_program(CMAKE_C_COMPILER NAMES x86_64-w64-mingw32-gcc)
find_program(CMAKE_CXX_COMPILER NAMES x86_64-w64-mingw32-g++)

### Minimal model for reproduction

This bug is independent of the model and code.

### Code required for reproduction

This bug is independent of the model and code.

### Confirmations

- [x] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [x] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

--- a/src/engine/engine_util_errmem.c
+++ b/src/engine/engine_util_errmem.c
@@ -99,7 +99,7 @@ void mju_writeLog(const char* type, const char* msg) {
 
 #if defined(_POSIX_C_SOURCE) || defined(__APPLE__) || defined(__STDC_VERSION_TIME_H__) || defined(__EMSCRIPTEN__)
     localtime_r(&rawtime, &timeinfo);
-#elif _MSC_VER
+#elif defined(_WIN32)
     localtime_s(&timeinfo, &rawtime);
 #elif __STDC_LIB_EXT1__
     localtime_s(&rawtime, &timeinfo);

--- a/src/user/user_vfs.cc
+++ b/src/user/user_vfs.cc
@@ -15,7 +15,7 @@
 #include "user/user_vfs.h"
 
 #include <sys/stat.h>
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
 #define stat _stat
 #endif
 

